backend/api/Services/QuestionTemplateService.cs

```diff
@@ -54,5 +55,70 @@ string supportNotes
             _context.SaveChanges();
             return newQuestionTemplate;
         }
+
+        public QuestionTemplate Edit(
+            QuestionTemplate questionTemplate,
+            Barrier barrier,
+            Organization organization,
+            string text,
+            string supportNotes,
+            Status status
+        )
+        {
+            if (questionTemplate == null)
+            {
+                throw new ArgumentNullException(nameof(questionTemplate));
+            }
+            QuestionTemplate newQuestionTemplate = new QuestionTemplate
+            {
+                Barrier = barrier,
+                Organization = organization,
+                Text = text,
+                SupportNotes = supportNotes,
+                Status = status,
+                Order = questionTemplate.Order,
+                previous = questionTemplate,
+            };
+            _context.QuestionTemplates.Add(newQuestionTemplate);
+
+            questionTemplate.Status = Status.Inactive;
+            _context.QuestionTemplates.Update(questionTemplate);
+            _context.SaveChanges();
+            return newQuestionTemplate;
+        }
+
+        public QuestionTemplate ReorderQuestionTemplate(QuestionTemplate questionTemplate, QuestionTemplate newNextQuestionTemplate = null)
+        {
+            List<QuestionTemplate> questionTemplates = _context.QuestionTemplates.OrderBy(qt => qt.Order).ToList();
+
+            int newOrder = newNextQuestionTemplate.Order - 1;
+            if (newNextQuestionTemplate.Equals(null))
+            {
+                newOrder = _context.QuestionTemplates.Max(qt => qt.Order);
+            }
```

Comment:
I am guessing this would give an error on the line above if the if is true. 

```suggestion
            int newOrder;
            if (newNextQuestionTemplate.Equals(null))
            {
                newOrder = _context.QuestionTemplates.Max(qt => qt.Order);
            }
            else 
            {
                newOrder = newNextQuestionTemplate.Order - 1;
            }
```

Reply:
yes, fixed